client.go

```diff
@@ -15,18 +15,20 @@ type CustomHTTPClient struct {
 	dedupeHashTable     *sync.Map
 	dedupeOptions       DedupeOptions
 	skipHTTPStatusCodes []int
+	errorChannel        chan error
```

Comment:
I think a shorter `errChan` might be better, but that's a personal preference. As you wish!